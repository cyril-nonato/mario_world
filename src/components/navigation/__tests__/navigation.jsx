import React from 'react'
import {cleanup} from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

import Navigation from '../navigation.component'
import renderWithRouter from '../../../utils/test_utils/renderWithRouter';

afterEach(cleanup);

describe('Navigation Component Test', () => {
  
  test('should match the snapshot', () => {
    const {asFragment} = renderWithRouter(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have the expected href attribute', async () => {
    const {getByText} = renderWithRouter(<Navigation />);

    expect(getByText('mario world')).toHaveAttribute('href', '/');
    expect(getByText('sign-in')).toHaveAttribute('href', '/sign-in');
    expect(getByText('sign-up')).toHaveAttribute('href', '/sign-up');
  })
});