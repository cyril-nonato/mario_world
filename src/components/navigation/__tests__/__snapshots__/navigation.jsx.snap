// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component Test should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 3.5rem;
  width: 100%;
  background: #2E2F2F;
}

.c1 {
  height: 100%;
  display: grid;
  width: 80%;
  margin: 0 auto;
  grid-template-columns: 2fr 4fr 1fr 1fr;
  grid-template-rows: 100%;
  font-size: 1.2rem;
  text-transform: uppercase;
}

.c2 {
  grid-column: 1 / 2;
  grid-row: 1 / 2;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
}

.c3 {
  font-size: 1.8rem;
  color: #FFFFFF;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c4 {
  grid-column: 3 / 4;
  grid-row: 1 / 2;
}

.c6 {
  grid-column: 4 / 5;
  grid-row: 1 / 2;
}

.c5 {
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c5:hover {
  background: #454545;
}

.c5.active {
  background: #454545;
}

<nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          aria-current="page"
          class="c3 active"
          href="/"
        >
          mario world
        </a>
      </li>
      <li
        class="c4"
      >
        <a
          class="c5"
          href="/sign-in"
        >
          sign-in
        </a>
      </li>
      <li
        class="c6"
      >
        <a
          class="c5"
          href="/sign-up"
        >
          sign-up
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
